lazy_theta_star_p_planner/src/lazy_theta_star_p_planner.cpp

```diff
@@ -0,0 +1,190 @@
+  if (!(planner_->isSafe(planner_->src.x,
+    planner_->src.y)) || !(planner_->isSafe(planner_->dst.x, planner_->dst.y)))
+  {
+//    RCLCPP_INFO(node_->get_logger(), "EITHER OF THE START OR GOAL POINTS ARE AN OBSTACLE! ");
+//    coordsW world{};
+//    planner_->costmap_->mapToWorld(planner_->src.x, planner_->src.y, world.x, world.y);
+//    path.push_back({world.x, world.y});
+//    planner_->costmap_->mapToWorld(planner_->dst.x, planner_->dst.y, world.x, world.y);
+//    path.push_back({world.x, world.y});
+//    global_path = linearInterpolation(path, interpolation_dist_);
+	global_path.poses.clear();
+  } else if (planner_->generatePath(path)) {
+    RCLCPP_INFO(node_->get_logger(), "RECEIVED THE PATH");
+    global_path = linearInterpolation(path, interpolation_dist_);
+  } else {
+    global_path.poses.clear();
+  }
+
+  auto stop = std::chrono::steady_clock::now();
+  auto dur = std::chrono::duration_cast<std::chrono::microseconds>(stop - start);
+  std::cout << "the time taken in microseconds is \t\t : " << dur.count() << '\n';
+  path.clear();
+  std::cout << global_path.poses.size() << '\n';
+  global_path.header.stamp = node_->now();
+  global_path.header.frame_id = globalFrame_;
+  RCLCPP_INFO(node_->get_logger(), "SENT OUT THE PATH");
+
+  RCLCPP_INFO(node_->get_logger(), "the cost at src : %i",
+    planner_->costmap_->getCost(planner_->src.x, planner_->src.y));
+  RCLCPP_INFO(node_->get_logger(), "the cost at dst : %i",
+    planner_->costmap_->getCost(planner_->dst.x, planner_->dst.y));
+}
+
+nav_msgs::msg::Path LazyThetaStarP_Planner::linearInterpolation(
+  const std::vector<coordsW> & raw_path,
+  const double & dist_bw_points)
+{
+  nav_msgs::msg::Path pa;
+
+  for (unsigned int j = 0; j < raw_path.size() - 1; j++) {
+    geometry_msgs::msg::PoseStamped p;
+    coordsW pt1 = {raw_path[j].x, raw_path[j].y};
+    p.pose.position.x = pt1.x;
+    p.pose.position.y = pt1.y;
+    pa.poses.push_back(p);
+
+    coordsW pt2 = {raw_path[j + 1].x, raw_path[j + 1].y};
+    geometry_msgs::msg::PoseStamped p1;
+    double distance = std::hypot(pt2.x - pt1.x, pt2.y - pt1.y);
+    int loops = static_cast<int>(distance / dist_bw_points);
+    double sin_alpha = (pt2.y - pt1.y) / distance;
+    double cos_alpha = (pt2.x - pt1.x) / distance;
+    for (int k = 1; k < loops; k++) {
+      p1.pose.position.x = pt1.x + k * dist_bw_points * cos_alpha;
+      p1.pose.position.y = pt1.y + k * dist_bw_points * sin_alpha;
+      pa.poses.push_back(p1);
+    }
+  }
+  return pa;
+}
+void LazyThetaStarP_Planner::setStartAndGoal(
+  const geometry_msgs::msg::PoseStamped & start,
+  const geometry_msgs::msg::PoseStamped & goal)
+{
+  unsigned int src_[2], dst_[2];
+  planner_->costmap_->worldToMap(start.pose.position.x, start.pose.position.y, src_[0], src_[1]);
+  planner_->costmap_->worldToMap(goal.pose.position.x, goal.pose.position.y, dst_[0], dst_[1]);
+
+  planner_->src = {static_cast<int>(src_[0]), static_cast<int>(src_[1])};
```

Comment:
`setGoal` and `setStart` functions would be good here so that you're not directly accessing variables of this object.